### ISSUE TITLE
Unable to link two Sphinx documents using :ref: roles

### ISSUE THREAD

**Issue by jgossage** [globalvillagesoftware/infrastructure#4]
We expect to make heavy use of this feature in the _Global Village_ documentation. It is not clear at the moment, whether this is simply an error on my part or a genuine bug and if a bug, what component the bug lives it.

If I have two files in the same document that I want to link where the target of the link is in the first file and the link is in the second file, I use the following code, lifted from the _Sphinx_ documentation:

File 1 contains:

```
.. ref _software-prerequisites:

***********
Major Tools
***********
```

which is supposed to create the target for a link.

File 2 contains:

```
Software Configuration
----------------------

:ref: `See <software-prerequisites>` for details of the software needed.
```

where a link is supposed to be generated. No diagnostic is reported by `sphinx-build`. The link is rendered as the following in the generated HTML:

```
<div class="section" id="software-configuration">
<h4>Software Configuration<a class="headerlink" href="#software-configuration" title="Permalink to this headline">¶</a></h4>
<dl class="field-list simple">
<dt class="field-odd">ref</dt>
<dd class="field-odd"><p><cite>See &lt;software-prerequisites&gt;</cite> for details of the software needed.</p>
</dd>
</dl>
</div>
```

This call results in the following text showing when the HTML is displayed in a browser:

`  `See <software-prerequisites>` for details of the software needed.`

I am unable to find **markdown** markup that will allow the line seen in the browser to be shown here exactly as it is seen in the browser. The entire link has been rendered as text by the Javascript in the `sphinx_rtd_theme` that is being used to display this document instead of creating a link. 

The following steps are needed to resolve this problem which could be a _Sphinx_ problem or a  _Read The Docs Theme_  problem:

* Examine the generated HTML to see why it behaves as it does. This will hopefully result in understanding who ought to own the problem.
* Check with the appropriate organization to see if they support this feature.
* If they say the feature is supported:
  * Raise an issue with them, reporting the bug.
  * Fix the bug locally and supply a _Pull Request_ to them with the bug fix.

* If they do not support the feature:
  * Raise an enhancement issue with them, requesting this support.
  * Fix the lack of support locally and supply a _Pull Request_ to them with the enhancement.
* Run the documents with the locally patched code.
* Remove the local fix when the organization publishes it if they do. 

The bottom line is that the problem needs to be fixed locally. This is not a problem since _Sphinx_ allows local HTML and Javascript to be supplied with each document.  If the problem is in the `sphinx-build` code, it may become necessary to use a local copy until the fix is published.

**Comment by jgossage** [comment]
On further examination, it appears that the generated HTML should do exactly what has happened. The problem now becomes to determine why `sphinx_build` generated such code.

**Comment by jgossage** [comment]
Another possibility is that `docutils` comes into the picture as a possible source of the problem. `docutils` provides low-level support to `Sphinx`.

**Comment by jgossage** [comment]
A workaround for this is to use standard Sphinx facilities for linking between different parts of a document.
